test_runner/regress/test_logical_replication.py

```diff
@@ -0,0 +1,54 @@
+import time
+from fixtures.log_helper import log
+from fixtures.neon_fixtures import NeonEnv
+
+def test_logical_replication(neon_simple_env: NeonEnv, vanilla_pg):
+    env = neon_simple_env
+
+    env.neon_cli.create_branch("test_logical_replication", "empty")
+    endpoint = env.endpoints.create_start("test_logical_replication")
+
+    log.info("postgres is running on 'test_logical_replication' branch")
+    pg_conn = endpoint.connect()
+    cur = pg_conn.cursor()
+
+    cur.execute("create table t(pk integer primary key, payload integer)")
+    cur.execute("create publication pub1 for table t")
+#    cur.execute("SELECT DISTINCT t.schemaname, t.tablename, t.attnames FROM pg_catalog.pg_publication_tables t WHERE t.pubname IN ('pub1')")
+#    time.sleep(1)
```

Comment:
Let's use `wait_until` or similar here and below instead of just sleeping, to avoid flakiness.

Reply:
We can not use `wait_until` here because we do not know which LSN we should wait for (it is logical replication!).
It may be possible to play with origins but brrr...

In principle we can add loop and perform several iteration waiting until proper result is returned.

Reply:
I now don't see why this particular sleep is needed at all. `create subscription sub1` should be fine immediately after creating publication.

As for the rest -- yes, instead of waiting for LSN we should just do e.g. `cur.execute("insert into t values (generate_series(1,1000), 0)")` and check for expected value in wait_until callback.

Reply:
> I now don't see why this particular sleep is needed at all. create subscription sub1 should be fine immediately after creating publication.

As far as I know `create subscription` (and also noticed many times in practice) doesn't wait until replication sync is performed. So quite often just after create of subscription the table is empty (especially if table is large).
But after some time all changes are delivered.

Reply:
Right, but what you are describing is fixed by next sleep/wait, not this one.

Reply:
O yes, sorry
This sleep and select were added because of the problem with request lsn in walproposer.
now it is not needed.

Reply:
I have added `logical_replication_sync` function to wait for sync between publisher and subscriber